grafana/grafana
# Canvas: Fix field not found

Updates metric value text when no data is returned. 
Fixes InfluxDB "Field not found" issue.

Before

https://github.com/grafana/grafana/assets/88068998/be41ca20-ea30-46ec-aae8-af1e7643b93c

After

https://github.com/grafana/grafana/assets/88068998/b4ba93b4-5972-4e46-9149-906a597e876a


Fixes #67580


**Special notes for your reviewer:**
- We should test it with other datasources also, not sure the best way to do that, but there seems to be some other inconsistencies - see #68414
- ^ There's another related issue with [zabbix](https://github.com/alexanderzobnin/grafana-zabbix/issues/1628)


Testing steps:
- Create a Canvas panel with InfluxDB as datasource
- Add a metric value element and pick a field for text
- The metric value should display the correct value
- If there's no data returned, should display "No data"



Please check that:
